horace_core/sqw/@main_header_cl/main_header_cl.m

```diff
@@ -130,50 +147,89 @@
             end
             obj.filename_ = val;
         end
-        %
+
         function tit = get.filepath(obj)
             tit = obj.filepath_;
         end
+
         function obj = set.filepath(obj,val)
             if ~(ischar(val) || isstring(val))
                 error('HORACE:main_header:invalid_argument', ...
-                    'filename has to be string or char array and it is %s',...
+                    'filepath has to be string or char array and it is %s',...
                     class(val));
             end
             obj.filepath_ = val;
         end
-        %
+
         function nf = get.nfiles(obj)
             nf = obj.nfiles_;
         end
+
         function obj = set.nfiles(obj,val)
-            if ~isnumeric(val) || val<0
+            if ~isnumeric(val) || val<0 || ~isscalar(val)
                 error('HORACE:main_header:invalid_argument', ...
-                    'Number of files should have non-negative numeric value. It is %s',...
+                    'Number of files should have a single non-negative numeric value. It is %s',...
```

Comment:
This is a bit unclear. Prefer "number of file should BE a single non-negative numeric value. Instead it is BEING SET TO %s...."

Reply:
This is an `error`, so it is not being set to anything, but ignored.